marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/mapping/getNewDocumentTest.sjs

```diff
@@ -0,0 +1,183 @@
+const test = require("/test/test-helper.xqy");
+const DataHubSingleton = require('/data-hub/5/datahub-singleton.sjs');
+const dataHub = DataHubSingleton.instance();
+const hubTest = require("/test/data-hub-test-helper.sjs");
+
+const jsonUri = '/test/sampleCustomerDoc.json';
```

Comment:
Should all test data be moved out of this file?

Reply:
Here's what I recommend:

1. Use JUnit tests against the DS-generated interface for the scenarios of - URI exists, and URI doesn't exist. 
1. Use ml-unit-test for all of the actual logic, but invoke a function that does all the logic once the result of cts.doc has been obtained

That allows you to write ml-unit-tests that don't persist anything, which leads to much simpler tests.

Reply:
I added the JUnit tests as recommended, which surfaced a couple errors with that HC endpoint 👍.